File: docs/examples/python/parent_process.py
from libmem import *

current_process = LM_GetProcess()
if current_process is None:
    print("[*] Failed to get current process")
    exit(-1)

print(f"[*] Process ID:          {current_process.pid}")
print(f"[*] Parent Process ID:   {current_process.ppid}")

parent_process = LM_GetProcessEx(current_process.ppid)
if parent_process is None:
    print("[*] Failed to get parent process")
    exit(-1)

print(f"[*] Parent Process Name: {parent_process.name}")

File: docs/examples/c/parent_process.c
#include <libmem/libmem.h>

int
main()
{
	lm_process_t current_process;
	lm_process_t parent_process;

	/* get the current process */
	if (!LM_GetProcess(&current_process)) {
		printf("[*] Failed to get current process\n");
		return -1;
	}

	printf("[*] Process ID:          %u\n", current_process.pid);
	printf("[*] Parent Process ID:   %u\n", current_process.ppid);

	/* get the parent process using the 'ppid' field */
	if (!LM_GetProcessEx(current_process.ppid, &parent_process)) {
		printf("[*] Failed to get the parent process\n");
		return -1;
	}

	printf("[*] Parent Process Name: %s\n", parent_process.name);

	return 0;
}
